Resources/public/js/recommendationsupport.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+(function (global, doc) {
+    var eZ = global.eZ = global.eZ || {};
+
+    /**
+     * Contains logic needed to fetch and display YooChoose recommendations
```

Comment:
>you don't need to generate unique variables names since you are in an anonymous function. But actually, given how the client and the renderer are used, instead of defining 2 classes, I would define 2 functions instead (fetchRecommendations(config, callback) and renderRecommendations(config) for instance)

I got you wrong...